courses/tasks.py

```diff
@@ -0,0 +1,58 @@
+"""
+Tasks for the courses app
+"""
+import logging
+from mitxpro.celery import app
+from mitxpro.utils import now_in_utc
+from courses.models import CourseRun, CourseRunCertificate
+from courses.utils import ensure_course_run_grade, process_course_run_grade_certificate
+from courseware.api import get_edx_grades_with_users
+
+log = logging.getLogger(__name__)
+
+
+@app.task
+def generate_course_certificates():
+    """
+    Task to generate certificates for courses.
+    """
+    log.info("Task generate_course_certificates: Executed")
+    now = now_in_utc()
+    course_runs = CourseRun.objects.filter(end_date__lt=now).exclude(
+        courseware_id__in=CourseRunCertificate.objects.values_list(
+            "course_run__courseware_id", flat=True
+        )
+    )
+
+    for run in course_runs:
+        edx_grade_user_iter = get_edx_grades_with_users(run)
+
+        for edx_grade, user in edx_grade_user_iter:
+            course_run_grade, created, updated = ensure_course_run_grade(
+                user=user, course_run=run, edx_grade=edx_grade
+            )
+
+            if created:
+                log.info("generate_course_certificates: New grade record created")
+            elif updated:
+                log.info(
+                    "generate_course_certificates: Grade record updated with new data from edX grades API"
+                )
+
+            _, created, deleted = process_course_run_grade_certificate(
+                course_run_grade=course_run_grade
+            )
+
+            if deleted:
+                log.warning(
+                    "generate_course_certificates: Certificate deleted as CourseRunGrade"
+                    " does not contain the grade detail for user % and course_run %",
+                    user,
+                    run,
+                )
+            else:
```

Comment:
`elif created:` because if the command is run again, both `created` and `deleted` will be `False` because the certificate would exist and would not be overwritten, for pretty much all certificates.